Universal-Connect-Project/ucw-app
# Implementation of a monorepo, using Turborepo

Uses https://turbo.build/ to help streamline development of the ucw-app. Resolves UCP-39.

_Note: Most of the 150+ file changes are due to moving files around, and running `lint --fix`_

### Tasks:

- [x] Implement Turborepo folder structure
- [x] Add necessary turborepo files
- [x] Update npm scripts to support turborepo conventions
- [x] Update Docker-related files
- [x] Update GH Action workflows (e2e/unit tests/env)
- [x] Fix lint issues
- [x] Update READMEs
- [x] Create new READMEs for monorepo and docker
- [x] Clean-up docker files for better efficiency
- [x] Update docker hub publish workflow
- [x] Fix husky to work with monorepo structure

### Ways to test:

After pullling this branch (all of this from the root of the project, unless specified): 

- Run `npm run clean`
- Run `npm ci`
- Run `npm run test`. All tests should pass.
- Run `npm run dev`. 
- Once that is running, run `ucw-example` from another terminal, and check the example app in a browser. It should load as expected.
- Kill the above process, but keep the example app running.
- Try running the example app again. It should fail to load the iframe contents.
- Next, run `docker compose up`. 
- Once that finishes, try the example app again. It should load as expected.
